### PR TITLE
doc: create the doc for the debuggable scheduler

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/area simulator
/kind cleanup
/kind documentation

#### What this PR does / why we need it:

- rename externalscheduler package to debuggablescheduler package.
- add the doc to describe the debuggable scheduler.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: simulator/pkg/debuggablescheduler/debuggable_scheduler.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	"k8s.io/kubernetes/pkg/scheduler/framework/runtime"
+
+	simulatorconfig "sigs.k8s.io/kube-scheduler-simulator/simulator/config"
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler"
+	simulatorschedulerconfig "sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler/config"
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler/plugin"
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/scheduler/storereflector"
+)
+
+// CreateOptionForOutOfTreePlugin creates the option which can be help with running the external scheduler.
```
Comment (196Ikuchil) [inline simulator/pkg/debuggablescheduler/debuggable_scheduler.go:33]:
```suggestion
// CreateOptionForOutOfTreePlugin creates the option which can be help with running the debuggable scheduler.
```
In nearly future, we may be replacing our simulator's internal scheduler with this debuggable scheduler. So this method aims not only external scheduler but also the internal one.

File: simulator/pkg/debuggablescheduler/debuggable_scheduler.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	if err != nil {
+		return nil, nil, xerrors.Errorf("creates a new Clientset for kubeconfig: %w", err)
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	if err := sharedStore.ResisterResultSavingToInformer(clientSet, ctx.Done()); err != nil {
+		return nil, cancel, xerrors.Errorf("ResisterResultSavingToInformer of sharedStore: %w", err)
+	}
+
+	// black magic: We need to use the scheduler config converted for the simulator in the external scheduler.
```
Comment (196Ikuchil) [inline simulator/pkg/debuggablescheduler/debuggable_scheduler.go:103]:
```suggestion
	// black magic: We need to use the scheduler config converted for the simulator in the debuggable scheduler.
```

**Review by 196Ikuchil — COMMENTED** [review]
Thanks! I left some small nits🙏

**Comment by sanposhiho** [comment]
I'll give an update after https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/302

**Comment by sanposhiho** [comment]
@196Ikuchil Updated!

**Review by 196Ikuchil — APPROVED** [review]
@sanposhiho 
Looks good 🚀 

/lgtm

**Comment by sanposhiho** [comment]
rebased 👌

File: README.md [hunk 3]
```diff
@@ -41,15 +41,22 @@ We have several components:
...
 You can create any resources by communicating with kube-apiserver in any ways (kubectl, k8s client library, or web UI described next)
 and see how your Pods are scheduled.
 
 When you create Pods, Pods will get annotations from the simulator which contains the scheduling results per plugins or extenders.
+=======
+You can create any resources by any ways (kubectl, k8s client library, or web UI described next).
+And when you create Pods, 
+Pods will be scheduled by the [debuggable scheduler](./simulator/docs/debuggable-scheduler.md),
+and they'll get the annotations that explain how each Pod was evaluated by each scheduler plugin.
+>>>>>>> 124eecc3 (doc: create the doc for the debuggable scheduler)
```
Comment (196Ikuchil) [inline README.md:59]: you may missed this 👀

**Review by 196Ikuchil — COMMENTED** [review]
@sanposhiho 
🙏

File: README.md (hunk 3 above)
Comment (sanposhiho) [reply README.md]: 🙏 🙏 🙏 🙏

**Review by 196Ikuchil — APPROVED** [review]
Thanks👍 

/lgtm

**Pull request merged**